### PR TITLE
Bugfix - AudioController issues

### PR DESCRIPTION
**Pull request by notexplosive**
## Summary
Fixes for #33 

On `AudioController.Start` we create a child game object and parent it to the AudioController, that GameObject has its own SoundSource that is used for playing SFX while the parent object's sound source is for Music.

### Before
- SFX volume was the same as music volume
- Songs wouldn't loop

### After
- SFX volume is now its own value separate from music volume
- Songs now loop

## Testing instructions
In the `AudioController - Test Scene` you now have separate volume sliders for SFX and Music. Press A and B to change tracks, press C to play a sound effect.

## Additional information

- `[ ]` Changes UI
- `[/]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource: 
  #33

### CODE REVIEW

File: Assets/Scripts/AudioController/AudioController.cs [hunk 1]
```diff
@@ -5,17 +5,27 @@ public class AudioController : MonoBehaviour, IAudioController
 {
```
Comment (JuiceGrape) [inline Assets/Scripts/AudioController/AudioController.cs:5]: Add a RequireComponent flag to the class to make sure it always has an audio source

File: Assets/Scripts/AudioController/AudioController.cs [hunk 2]
```diff
@@ -32,8 +42,13 @@ void Start()
 
         _musicFader = new MusicFader();
 
-        _audioSource = GetComponent<AudioSource>();
-        Debug.Assert(_audioSource != null, "AudioController is missing AudioSource Component");
+        _musicAudioSource = GetComponent<AudioSource>();
+        Debug.Assert(_musicAudioSource != null, "AudioController is missing AudioSource Component");
+        _musicAudioSource.loop = true;
+
+        var sfxGameObject = new GameObject("SFX Player");
```
Comment (JuiceGrape) [inline Assets/Scripts/AudioController/AudioController.cs]: Why is only the SFX source a child? Probably best to make them both a child and turn this into a function

**Review by JuiceGrape — APPROVED** [review]

File: Assets/Scripts/AudioController/AudioController.cs (hunk 2 above)
Comment (JuiceGrape) [reply Assets/Scripts/AudioController/AudioController.cs]: In which case the other comment is no longer applicable

File: Assets/Scripts/AudioController/AudioController.cs [hunk 3]
```diff
@@ -32,8 +42,13 @@ void Start()
...
+        sfxGameObject.transform.parent = this.transform;
+        _sfxAudioSource = sfxGameObject.AddComponent<AudioSource>();
```
Comment (TooManyMoths) [inline Assets/Scripts/AudioController/AudioController.cs:51]: Suggestion: Use "new GameObject("SFX Player", typeof(AudioSource));" to create a new game object with an audio source.

**Review by PhoebeMitchell — APPROVED** [review]
Tested in editor. Sound effects play when transitioning songs, and songs loop.

**Pull request merged**